### PR TITLE
Redeem page and components

### PR DESCRIPTION
**Pull request by sophiacodes**
🥳 **Adds**

 - Coupon onboarding redeem page and components

### CODE REVIEW

File: src/util/helpers/truncateSignature.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import Web3 from 'web3';
+
+export const isSignatureValid = (signature: string) =>
```
Comment (jtrein) [inline src/util/helpers/truncateSignature.ts:3]: We could probably just import `import {isHex} from 'web3-utils'` and use it directly in the `truncateSignature`.

**Review by jtrein — CHANGES_REQUESTED** [review]
Just looked at this briefly. Overall, nice! Added a couple small possible changes.

Also, if you could open an issue for writing tests for this, we should do that to keep up to standard with the rest of the Dapp.

File: src/util/helpers/truncateSignature.ts (hunk 1 above)
Comment (sophiacodes) [reply src/util/helpers/truncateSignature.ts]: good spot! done

File: src/components/feedback/LoaderWithEmoji.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    return function cleanup() {
+      timeoutId && clearTimeout(timeoutId);
+    };
+  }, [showAfterMs]);
+
+  return !showAfterMs || (showAfterMs && showLoaderAfterMs) ? (
+    <div className="loader--emoji">
+      <span role="img" aria-label="Loading content...">
+        🎟
```
[the remark this answers is not in the document]
Comment (sophiacodes) [reply src/components/feedback/LoaderWithEmoji.tsx:26]: Yeah I'll create an issue for the tests and update to pass a prop

File: src/components/feedback/LoaderWithEmoji.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const timeoutId =
+      showAfterMs && setTimeout(() => setShowLoaderAfterMs(true), showAfterMs);
+
+    return function cleanup() {
+      timeoutId && clearTimeout(timeoutId);
+    };
+  }, [showAfterMs]);
+
+  return !showAfterMs || (showAfterMs && showLoaderAfterMs) ? (
+    <div className="loader--emoji">
```
Comment (jdville03) [inline src/components/feedback/LoaderWithEmoji.tsx:25]: I don't think this className exists in the `_loader.scss` stylesheet (was removed before). May need to add back the needed styles for this if necessary.

**Review by jdville03 — APPROVED** [review]

**Pull request merged**